src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -178,4 +187,66 @@ protected CustomerInvoiceDetail createSourceAccountingLinesByContractControlAcco
                 coaCode, accountNumber, objectCode, getTotalAmountForInvoice(contractsGrantsInvoiceDocument), 1);
     }
 
+    /**
+     * Overridden to also update the invoice due date.
+     * 
+     * NOTE: There isn't an easy way to hook in the invoice-due-date updates independently of this
+     * without overlaying the CINV document class. (This method only appears to be called when CINV docs
+     * enter PROCESSED status, so hooking in here is okay for now.) If ContractsGrantsInvoiceDocument
+     * gets updated to make it easier to hook in custom PROCESSED-status handling, then we should rework
+     * this customization.
```

Comment:
Should we create a kualico ticket with suggested changes to ContractsGrantsInvoiceDocument?  If so, we should also create a follow user story for us to rework this

Reply:
Ok, I'll create a PTS ticket for us to follow up with KualiCo to make this part easier to customize.

Reply:
I've created KFSPTS-15550 to keep track of the CINV override limitations.